File: projects/Hood/Models/ComplexTypes/Response.cs
using System.Collections.Generic;
using Microsoft.AspNetCore.Identity;
using Hood.Interfaces;
using Hood.Extensions;
using System;

namespace Hood.Models
{
    public class Response
    {
        public Array Data { get; set; }
        public int Count { get; set; }
        public string Errors { get; set; }
        public string Message { get; set; }
        public bool Success { get; set; }
        public string Url { get; set; }
        public Exception Exception { get; set; }
        public IMediaObject Image { get; set; }

        public Response(Array data, int count, string message = "Succeeded!")
        {
            Success = true;
            Data = data;
            Message = message;
            Count = count;
        }

        public Response(string errors, string message = "An error occurred!")
        {
            Success = false;
            Message = message;
            Errors = errors;
        }

        public Response(bool success, string message = "Succeeded!")
        {
            Success = success;
            Message = message;
            Errors = message;
        }
        public Response(bool success, IMediaObject media, string message = "Succeeded!")
        {
            Success = success;
            Message = message;
            Errors = message;
            Image = media;
        }

        public Response(IEnumerable<IdentityError> errors, string message = "An error occurred!")
        {
            Success = false;
            Message = message;
            Errors = "";
            foreach (IdentityError err in errors)
            {
                Errors += err.Description + Environment.NewLine;

            }
        }
        public Response(Exception ex, string message = "An error occurred!")
        {
            Errors = ex.Message;
            Message = message;
            Success = false;
            Exception = ex;
        }
    }
}

File: projects/Hood.Development/src/ts/core/response.ts
import { KeyValue } from "../interfaces/KeyValue";
import { Alerts } from "./Alerts";
import { MediaObject } from "./Media";

export declare interface Response {

    data: any[];
    count: number;

    url: string;

    title: string;
    message: string;

    success: boolean;

    exception: KeyValue<string, string>[];
    errors: string;
    error: string;

    media: MediaObject;

    process(autoHide: number): void;

}

export class Response {
    constructor() {

    }

    static process(data: Response | any, autoHide: number = 5000) {
        if (data.success) {
            Alerts.success(data.message, data.title, autoHide);
        } else {
            Alerts.error(data.message, data.title, autoHide);
        }
    }
}

